### DISCUSSION TITLE
2.9.22 to 2.10.xx Upgrade steps?

### DISCUSSION THREAD

**Discussion by Ja50n-Kru** [NginxProxyManager/nginx-proxy-manager#3205] [Q&A]
Hi All, I am running NPM in a docker container, on Diet-Pi. Some months ago an upgrade from 2.9.22 to 2.10.xx resulted in 'Bad Gateway' and unable to do anything. I couldn't work out how to fix it, so went back to 2.9.22.

Is there a way to upgrade to 2.10.xx and not get the bad gateway error? I have Portainer, so would by default use that to recreate the container at latest. 

I've done vast amounts of searching, and gather the error is something to do with how the database was created on previous versions? I'm in over my head here and have no clue at all - any help appreciated!

**Comment by MickMorley** [comment]
For what its worth, I just upgraded from 2.9.19 to 2.10.4 today without an issue.  All I did was zip up my directory like instructed in case anything went wrong.  I am running a sqlite db.

**Comment by Ja50n-Kru** [comment]
Worked for a while, but I still get bad gateway if the docker or device restarts. 

Is there a step-by-step for thicko's how to migrate from the previously working config with the old compose file, to the new one without losing any data? I really mean step-by-step. 'Just change your docker compose yaml to the new one' can't be the answer - I must have to move existing data or something?

This is my current docker compose that worked flawlessly for 2.9.22 and before:

version: '3'
services:
  app:
    image: 'jc21/nginx-proxy-manager:2.9.22'
    restart: unless-stopped
    ports:
      - '80:80'
      - '81:81'
      - '443:443'
    environment:
      DB_MYSQL_HOST: 'db'
      DB_MYSQL_PORT: 3306
      DB_MYSQL_USER: 'npm'
      DB_MYSQL_PASSWORD: 'xxxxx'
      DB_MYSQL_NAME: 'npm'
    volumes:
      - ./data:/data
      - ./letsencrypt:/etc/letsencrypt
  db:
    image: 'jc21/mariadb-aria:latest'
    restart: unless-stopped
    environment:
      MYSQL_ROOT_PASSWORD: 'xxxxx'
      MYSQL_DATABASE: 'npm'
      MYSQL_USER: 'npm'
      MYSQL_PASSWORD: 'xxxxx'
    volumes:
      - ./data/mysql:/var/lib/mysql

**Reply by ipalienko** [reply]
Удалось решить проблему?
